connect/runtime/src/test/java/org/apache/kafka/connect/runtime/WorkerTaskTest.java

```diff
@@ -78,155 +74,80 @@ public void tearDown() {
+                retryWithToleranceOperator, Time.SYSTEM, statusBackingStore) {
 
-        workerTask.initialize(TASK_CONFIG);
-        EasyMock.expectLastCall();
-
-        workerTask.close();
-        EasyMock.expectLastCall();
+            @Override
+            public void initializeAndStart() {
+                fail("This method is expected to not be invoked");
+            }
 
-        replay(workerTask);
+            @Override
+            public void execute() {
+                fail("This method is expected to not be invoked");
+            }
+        };
 
         workerTask.initialize(TASK_CONFIG);
         workerTask.stop();
         workerTask.awaitStop(1000L);
 
         // now run should not do anything
         workerTask.run();
-
-        verify(workerTask);
     }
 
     @Test
     public void cancelBeforeStopping() throws Exception {
         ConnectorTaskId taskId = new ConnectorTaskId("foo", 0);
-
-        WorkerTask workerTask = partialMockBuilder(WorkerTask.class)
-                .withConstructor(
-                        ConnectorTaskId.class,
-                        TaskStatus.Listener.class,
-                        TargetState.class,
-                        ClassLoader.class,
-                        ConnectMetrics.class,
-                        RetryWithToleranceOperator.class,
-                        Time.class,
-                        StatusBackingStore.class
-                )
-                .withArgs(taskId, statusListener, TargetState.STARTED, loader, metrics,
-                        retryWithToleranceOperator, Time.SYSTEM, statusBackingStore)
-                .addMockedMethod("initialize")
-                .addMockedMethod("initializeAndStart")
-                .addMockedMethod("execute")
-                .addMockedMethod("close")
-                .createStrictMock();
-
         final CountDownLatch stopped = new CountDownLatch(1);
-        final Thread thread = new Thread(() -> {
-            try {
-                stopped.await();
-            } catch (Exception e) {
-            }
-        });
-
-        workerTask.initialize(TASK_CONFIG);
-        EasyMock.expectLastCall();
-
-        workerTask.initializeAndStart();
-        EasyMock.expectLastCall();
-
-        workerTask.execute();
-        expectLastCall().andAnswer(() -> {
-            thread.start();
-            return null;
-        });
 
-        statusListener.onStartup(taskId);
-        expectLastCall();
+        WorkerTask workerTask = new TestWorkerTask(taskId, statusListener, TargetState.STARTED, loader, metrics,
+                retryWithToleranceOperator, Time.SYSTEM, statusBackingStore) {
 
-        workerTask.close();
-        expectLastCall();
-
-        // there should be no call to onShutdown()
+            @Override
+            public void execute() {
+                try {
+                    stopped.await();
+                } catch (InterruptedException e) {
+                    Thread.currentThread().interrupt();
```

Comment:
We should never get interrupted during the test; probably safer to fail here:
```suggestion
                    fail("Unexpectedly interrupted during test");
```